internal/awsnuke/client_integration_test.go

```diff
@@ -0,0 +1,89 @@
+package awsnuke_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/internal/awsnuke"
+)
+
+var AWSAccountId = "353720269506"
+
+func TestDestroyEC2Instances(t *testing.T) {
+	createNewInstance(t, "us-east-1", "ami-6057e21a")
+
+	nuke := awsnuke.New(AWSAccountId)
+	instances, err := nuke.ListNonProtectedEC2Instances()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	err = nuke.DestroyEC2Instances(instances)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	instances, err = nuke.ListNonProtectedEC2Instances()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	if expect, actual := 0, len(instances); expect != actual {
+		t.Errorf("expected all instances to have been destroyed")
+	}
+}
+
+func createNewInstance(t *testing.T, region, ami string) {
+	ec2client := newEC2Client(region)
+	input := &ec2.RunInstancesInput{
+		ImageId:      aws.String(ami),
+		InstanceType: aws.String("t1.micro"),
+		MaxCount:     aws.Int64(1),
+		MinCount:     aws.Int64(1),
+	}
+	reservation, err := ec2client.RunInstances(input)
+	if err != nil {
+		t.Fatalf("failed to create ec2 instance: %v", err)
+	}
+
+	if len(reservation.Instances) < 1 {
+		t.Fatal("failed to create ec2 instance")
+	}
+
+	instance := reservation.Instances[0]
+	state := "running"
+OUTER:
```

Comment:
Instead of this, see [WaitUntilInstanceRunning](http://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.WaitUntilInstanceRunning).